crates/ethportal-api/src/types/content_value/beacon.rs

```diff
@@ -482,28 +482,23 @@ pub struct ForkVersionedHistoricalSummariesWithProof {
     pub historical_summaries_with_proof: HistoricalSummariesWithProof,
 }
 
-impl ForkVersionedHistoricalSummariesWithProof {
-    pub fn encode(&self) -> Vec<u8> {
-        let fork_digest = self.fork_name.as_fork_digest();
+impl Encode for ForkVersionedHistoricalSummariesWithProof {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
 
-        let mut data = fork_digest.to_vec();
-        data.extend(self.historical_summaries_with_proof.as_ssz_bytes());
-        data
+    fn ssz_append(&self, buf: &mut Vec<u8>) {
+        self.fork_name.as_fork_digest().ssz_append(buf);
+        self.historical_summaries_with_proof.ssz_append(buf);
```

Comment:
```suggestion
    fn ssz_append(&self, buf: &mut Vec<u8>) {
        buf.extend(&self.fork_name.as_fork_digest());
        self.historical_summaries_with_proof.ssz_append(buf);
```

the fork digest is just supposed to be appended not ssz encoded right? or do this encode to the same thing or something?